packages/u/usysconf-epoch/files/epoch.sh

```diff
@@ -0,0 +1,158 @@
+
+STATE_DIR="${STATE_DIR:=/var/solus/usr-merge}"
+MERGE_FLAG_FILE="${MERGE_FLAG_FILE:=${STATE_DIR}/merge-complete}"
+
+OLD_REPO="https://cdn.getsol.us/repo/shannon/eopkg-index.xml.xz"
+NEW_REPO="https://cdn.getsol.us/repo/shannon/eopkg-index.xml"
+
+export LC_ALL=C
+
+declare -A DESKTOP_FLAG_FILES=(
+    ["budgie"]="/usr/bin/budgie-desktop"
+    ["gnome"]="/usr/bin/gnome-shell"
+    ["kde"]="/usr/bin/startplasma-wayland"
+    ["xfce"]="/usr/bin/xfce4-session"
+)
+declare -A DESKTOP_SOFTWARE_CENTRE=(
+    ["budgie"]="discover"
+    ["gnome"]="gnome-software"
+    ["kde"]="discover"
+    ["xfce"]="discover"
+)
+declare -A SC_DESKTOP_FILES=(
+    ["discover"]="/usr/share/applications/org.kde.discover.desktop"
+    ["gnome-software"]="/usr/share/applications/org.gnome.Software.desktop"
+)
+
+function update_repo() {
+    local repo=""
+    local active=""
+    while read -r line
+    do
+        if [[ "${repo}" == "" ]]
+        then
+            repo="${line% *}"
+            active="${line##"${repo}" }"
+            continue
+        fi
+
+        url="${line##   }"
+        if [[ "${url}" == "${OLD_REPO}" ]]
+        then
+            echo "Updating repo ${repo} to ${NEW_REPO}"
+
+            # Workaround for eopkg #149 while repos are the same.
+            # Remove this when the repo is fixed.
+            eopkg add-repo -y --no-fetch "${repo}" "${NEW_REPO}"
+            eopkg add-repo -y "${repo}" "${NEW_REPO}.xz"
+
+            if [[ "${active}" == "[inactive]" ]]
+            then
+                eopkg disable-repo "${repo}"
+            fi
+        fi
+
+        repo=""
+    done <<< "$(eopkg list-repo --no-color)"
+}
+
+function is_installed() {
+    eopkg info --no-color --short "${1}" | grep -P '^\[inst\]' >/dev/null
+}
+
+function desktop_name() {
+    local field
+
+    if [[ "$2" != "" ]]
+    then
+        field="^Name\[$2\]="
+    else
+        field="^Name="
+    fi
+
+    local name
+    name="$(grep "${field}" "${SC_DESKTOP_FILES[$1]}" | head -n1 || true)"
+
+    echo "${name#*=}"
+}
+
+function sc_name() {
+    local name
+
+    for lang in "${LANG%.*}" "${LANG%_*}" ""
+    do
+        name="$(desktop_name "$1" "${lang}")"
+        if [[ "${name}" != "" ]]
+        then
+            echo "${name}"
+            return
+        fi
+    done
+}
+
+# Based on StackOverflow post by Fabio A.
+# See https://stackoverflow.com/a/49533938
+function _notify-send() {
+    local user
+
+    for bus in /run/user/*/bus
+    do
+        user="$(stat -c '%U' "${bus}")"
+        sudo -u "${user}" env DBUS_SESSION_BUS_ADDRESS="unix:path=$bus" notify-send "$@"
+    done
+}
+
+if [ ! -e "${MERGE_FLAG_FILE}" ]
+then
+    echo "Not ready: not usr-merged"
+    exit 1
+fi
+
+if [ "$(readlink /usr/bin/eopkg)" != "eopkg.bin" ]
+then
+    echo "Not ready: eopkg is not python3"
+    exit 1
+fi
+
+desktop=""
+removed_sc=false
+
+# Install appropriate software centre
+for d in "${!DESKTOP_FLAG_FILES[@]}"
+do
+    if [ -e "${DESKTOP_FLAG_FILES[$d]}" ]
```

Comment:
I believe that `[` actually resolves to a call to the coreutils `[` executable.

Given the script explicitly calls bash, perhaps using the `[[ ]]` bash built-ins would be useful?